homeassistant/components/blink/config_flow.py

```diff
@@ -94,23 +91,36 @@ async def async_step_user(self, user_input=None):
 
     async def async_step_2fa(self, user_input=None):
         """Handle 2FA step."""
+        errors = {}
         if user_input is not None:
             pin = user_input.get(CONF_PIN)
-            if await self.hass.async_add_executor_job(
-                self.blink.login_handler.send_auth_key, self.blink, pin
-            ):
-                return await self.async_step_user(user_input=self.data)
+            try:
+                if await self.hass.async_add_executor_job(
+                    _send_blink_2fa_pin, self.auth, pin
+                ):
+                    return await self.async_step_finish()
+                errors["base"] = "invalid_access_token"
+            except BlinkSetupError:
+                errors["base"] = "cannot_connect"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
 
         return self.async_show_form(
             step_id="2fa",
             data_schema=vol.Schema(
                 {vol.Optional("pin"): vol.All(str, vol.Length(min=1))}
             ),
+            errors=errors,
         )
 
-    async def async_step_import(self, import_data):
-        """Import blink config from configuration.yaml."""
-        return await self.async_step_user(import_data)
+    async def async_step_import(self, entry_data):
+        """Perform import of old entries."""
+        return await self.async_step_user(entry_data)
+
+    async def async_step_finish(self):
```

Comment:
This doesn't have to be a coroutine since we don't await inside.
```suggestion
    @callback
    def _async_finish_flow(self):
```